met-api/src/met_api/services/user_service.py

```diff
@@ -28,13 +33,60 @@ def create_or_update_user(self, user: dict):
 
         external_id = user.get('external_id')
         db_user = UserModel.get_user_by_external_id(external_id)
+
         if db_user is None:
             is_staff_user = user.get('identity_provider', '').lower() == LoginSource.IDIR.value
             access_type = UserType.STAFF.value if is_staff_user else UserType.PUBLIC_USER.value
             user['access_type'] = access_type
+            if len(user.get('roles', [])) == 0:
+                self._send_access_request_email(user)
             return UserModel.create_user(user)
+
         return UserModel.update_user(db_user.id, user)
 
+    @staticmethod
+    def _send_access_request_email(user: UserSchema) -> None:
+        """Send a new user email.Throws error if fails."""
+        to_email_address = current_app.config.get('ACCESS_REQUEST_EMAIL_ADDRESS', None)
+        if to_email_address is None:
+            return
+
```

Comment:
throw an error may be if to email is null ;its a necessary thing rite?

Reply:
its one option, I decided that if not setup the email part was disabled, but I think you are right